src/GraphQL.MicrosoftDI/ScopedResolveFieldContextAdapter.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using GraphQL.Execution;
+using GraphQL.Instrumentation;
+using GraphQL.Language.AST;
+using GraphQL.Types;
+
+namespace GraphQL.MicrosoftDI
+{
+    internal sealed class ScopedResolveFieldContextAdapter<TSource> : ScopedResolveFieldContextAdapter, IResolveFieldContext<TSource>
```

Comment:
Maybe make one class for both adapters - `ScopedResolveFieldContextAdapter<TSource>`?

Reply:
`ScopedFieldResolver<TReturnType>` then will use ` return resolver(new ScopedResolveFieldContextAdapter<object>(context, scope.ServiceProvider));`

Reply:
Ok